packages/vector_graphics_compiler/lib/src/geometry/basic_types.dart

```diff
@@ -104,6 +104,18 @@ class Rect {
     );
   }
 
+  /// Whether or not the rect has any contents.
+  bool get isEmpty => width == 0 || height == 0;
+
+  /// Whether or not [other] intersect this rectangle.
+  bool intersects(Rect other) {
+    if (isEmpty || other.isEmpty) {
+      return false;
+    }
+    return math.max(left, other.left) < math.min(right, other.right) &&
+        math.max(top, other.top) < math.min(bottom, other.bottom);
```

Comment:
Consider adding a comment that this method only works for sorted rects. We may also want some kind of method to sort rects. So far it hasn't been a problem but at some point we'll get a rect where this.bottom < this.top for example.

Alternatively, we just assert in the ctor that it's sorted...

Reply:
I added an assert and comment here for now